numpyro/contrib/hsgp/approximation.py

```diff
@@ -21,26 +21,22 @@
 import numpyro.distributions as dist
 
 
-def _non_centered_approximation(
-    phi: ArrayLike, spd: ArrayLike, m: int | list[int]
```

Comment:
why Array is used instead of ArrayLike?

Reply:
because `ArrayLike` can be a scalar (see https://jax.readthedocs.io/en/latest/jax.typing.html) but we want them to be true arrays